lib/validates_overlap/overlap_validator.rb

```diff
@@ -98,11 +102,14 @@ def generate_overlap_sql_conditions(record)
     if record.new_record?
       self.sql_conditions = main_condition
     else
-      self.sql_conditions = "#{main_condition} AND #{record_table_name(record)}.#{primary_key_name} != #{record.send(primary_key_name)}"
+      if primary_key_value(primary_key_name, record).is_a?String
```

Comment:
Whoa. Hey @dthyresson. This seems like a better way to do things. But to do this they had to grab  the columns off the model, which is passed in during initialization. Which I think would change the code more then we want to.